tests/submodule/add.c

```diff
@@ -128,3 +128,51 @@ void test_submodule_add__url_relative_to_workdir(void)
 
 	assert_submodule_url("TestGitRepository", git_repository_workdir(g_repo));
 }
+
+void test_submodule_add__path_exists_in_index(void)
```

Comment:
There's one problem with that test. You want to verify that creation of a submodule fails if the path exists in the index, but actually when you create your submodule it exists both inside the index and also inside of the working directory. So us failing here might not be the result from the entry existing in the index, but instead that file existing in the working directory.

You could either just remove the directory and file after you've added them to the index or, which is the preferred method if you ask me, instead add the index entry directly by using `git_index_add`.

Reply:
I get the problem you are describing, but I am unsure how to use `git_index_add`. How can I create a `git_index_entry` properly? I s there something I am missing? Other than `index_create_entry`, which is not exposed outside of `index.c` I do not know how to create an entry. For now, I am updating the routine to just delete the file and directory after I add to the index.

Reply:
Take a look at for example "tests/index/add.c", `test_add_entry()`. You can create an index entry by simply assigning the structure fields.